provider/server/exec_test.go

```diff
@@ -68,5 +67,69 @@ var _ = Describe("Exec", func() {
 		Expect(parsedResUrl.Host).To(Equal(parsedBaseURL.Host))
 		Expect(parsedResUrl.Scheme).To(Equal(parsedBaseURL.Scheme))
 		Expect(parsedResUrl.Path).To(MatchRegexp(`/exec/[^/?&]{8}`))
+
+		By("issuing exec with response URL received and verifying tty stream")
+		err = runExec(ctx, parsedResUrl)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Verifying same token cannot be used twice")
+		err = runExec(ctx, parsedResUrl)
+		Expect(err).To(MatchError(ContainSubstring("404 page not found")), "Rejecting unknown / expired token")
+
+		By("getting exec-url with new token")
+		execResp, err = machineClient.Exec(ctx, &iri.ExecRequest{MachineId: createResp.Machine.Metadata.Id})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("inspecting the result")
+		parsedResUrl, err = url.ParseRequestURI(execResp.Url)
+		Expect(err).NotTo(HaveOccurred(), "url is invalid: %q", execResp.Url)
+
+		By("deleting the machine")
+		_, err = machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
+		Expect(err).ShouldNot(HaveOccurred())
+		Eventually(func() bool {
+			_, err = libvirtConn.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id))
+			return libvirt.IsNotFound(err)
+		}).Should(BeTrue())
+
+		By("issuing exec for non existing machine")
+		err = runExec(ctx, parsedResUrl)
+		machineNotSynchErr := fmt.Sprintf("machine %s has not yet been synced", createResp.Machine.Metadata.Id)
+		Expect(err).To(SatisfyAny(MatchError(ContainSubstring("404 page not found")), MatchError(ContainSubstring(machineNotSynchErr))))
+
 	})
 })
+
+func runExec(ctx context.Context, execUrl *url.URL) error {
```

Comment:
this function doesn't try send any command to console or read output. So you don't test whole logic of Streaming server. You test just http server logic but i recommend test whole streaming logic with send command and read output. Maybe login of user will enough.

Reply:
It was not mentioned in the scope of issue https://github.com/ironcore-dev/libvirt-provider/issues/90 If really i can cover. It talks about only negative cases.

Reply:
we can discuss it on daily. But implement whole logic can help us catch error during development later.

Reply:
We are using go routine with infinite loop to read and write to stream, even when i am sending Escape Sequence to terminate the stream it is not coming out of the go routine. Tried with bytes.Buffer Read and Write methods, somehow escape sequence is not working with this kind of buffer. I have created separate issue to work on it further #207